app/frontend/src/features/profile/view/leaveReference/formSteps/submit/SubmitReference.tsx

```diff
@@ -0,0 +1,134 @@
+import { useMediaQuery, useTheme } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import { REFERENCE_SUCCESS, SUBMIT } from "features/profile/constants";
+import {
+  useWriteFriendReference,
+  useWriteHostReference,
+} from "features/profile/hooks/referencesHooks";
+import ReferenceOverview from "features/profile/view/leaveReference/formSteps/submit/ReferenceOverview";
+import {
+  ReferenceContextFormData,
+  useReferenceStyles,
+} from "features/profile/view/leaveReference/ReferenceForm";
+import { User } from "pb/api_pb";
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useParams } from "react-router";
+import { referenceTypeRoute } from "routes";
+import {
+  WriteFriendReferenceInput,
+  WriteHostRequestReferenceInput,
+} from "service/references";
+
+export interface SubmitReferenceProps {
+  user: User.AsObject;
+  referenceData: ReferenceContextFormData;
+}
+
+export default function SubmitReference({
+  user,
+  referenceData,
+}: SubmitReferenceProps) {
+  const {
+    writeFriendReference,
+    status: friendReferenceWritingStatus,
+    reset: resetFriendReferenceWriting,
+  } = useWriteFriendReference(user.userId);
+  const {
+    writeHostRequestReference,
+    status: hostRequestReferenceWritingStatus,
+    reset: resetHostRequestReferenceWriting,
+  } = useWriteHostReference(user.userId);
+
+  const { referenceType, hostRequest } = useParams<{
+    referenceType: string;
+    hostRequest?: string;
+  }>();
+  const theme = useTheme();
+  const classes = useReferenceStyles();
+  const isSmOrWider = useMediaQuery(theme.breakpoints.up("sm"));
+  const { handleSubmit } = useForm<ReferenceContextFormData>();
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const onFriendReferenceSubmit = () => {
+    const formData: WriteFriendReferenceInput =
+      referenceData.wasAppropriate === "true"
+        ? {
+            toUserId: user.userId,
+            wasAppropriate: true,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          }
+        : {
+            toUserId: user.userId,
+            wasAppropriate: false,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          };
+
+    resetFriendReferenceWriting();
+    writeFriendReference({
+      referenceData: formData,
+    });
```

Comment:
In the second argument to this function, you can do `{ onSuccess: () => history.push(...) }` and redirect to their reference page. Friend references are visible immediately.

Reply:
Done, however redirecting to `/user/:username/references` does not open the References tab, just the user's profile. That's probably fine for now?